Removed locationButton under certain conditions.

daika7ana opened the pull request:
## Description
* Removed locationButton when Geolocation is toggled on, because current
position = worker position anyhow.

## Motivation and Context
<!--- Why is this change required? What problem does it solve? -->
<!--- If it fixes an open issue, please link to the issue here. -->

## How Has This Been Tested?
Tested on Win10: Edge, Chrome, Firefox.
Tested on Android 6.0.1: Chrome.

## Screenshots (if appropriate):
![before](https://cloud.githubusercontent.com/assets/11703528/17201454/3caa7d7c-5497-11e6-81c9-a6f4757a334b.png)
![after](https://cloud.githubusercontent.com/assets/11703528/17201456/44313ebe-5497-11e6-9072-8ce6195dc96f.png)

## Types of changes
<!--- What types of changes does your code introduce? Put an `x` in all the boxes that apply: -->
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:
<!--- Go over all the following points, and put an `x` in all the boxes that apply. -->
<!--- If you're unsure about any of these, don't hesitate to ask. We're here to help! -->
- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

georgo commented on static/dist/js/map.js:
```diff
@@ -1237,6 +1240,10 @@ $(function() {
 
   $('#geoloc-switch').change(function() {
+    if (!Store.get('geoLocate'))
+		document.getElementById("location-button").style.display = 'none'; 
```
I would recommend use $("#location-button").hide() instead of this.

georgo commented on static/dist/js/map.js:
```diff
@@ -1237,6 +1240,10 @@ $(function() {
+    if (!Store.get('geoLocate'))
+		document.getElementById("location-button").style.display = 'none'; 
+	else 
+		document.getElementById("location-button").style.display = 'inline'; 
```
I would recommend use $("#location-button").show() instead of this.

andrewda replied:
Use the standard coding conventions, please. Use jQuery, as @georgo pointed out, and add curly braces. Also, indent with two spaces. We are going to be strictly enforcing these rules.

willgrnr commented:
+1

daika7ana commented:
Cheers for everything that you've said and for the input :+1:  Striving to be better at this. Fixed everything that you've pointed out, retested and rebased. How does it look?

andrewda commented on static/dist/js/map.js:
```diff
@@ -1237,6 +1241,12 @@ $(function() {
+    if (!Store.get('geoLocate')) {
+      $("#location-button").hide();
+    }
+    else {
```
Hate to be so nit-picky, but please make the closing curly brace and the `else` statement on the same line. We're trying to keep and maintain a code standard.

georgo commented on static/dist/js/map.js:
```diff
@@ -1128,6 +1132,11 @@ $(function () {
   });
 
   $('#geoloc-switch').change(function () {
+    if (!Store.get('geoLocate')) {
```
This negative condition doesn't make sense (because on the end of this function it'll probably change and it will be confusing.
Use ```this.checked``` instead. ;)

daika7ana replied:
I initially did this, however for some reason this was not working on android default browser. However that being said, now the map doesn't work at all in the default browser but it works perfectly fine in Chrome/Firefox. So I guess the problem lies elsewhere.

AHAAAAAAA commented:
This has already been merged in it seems, but thanks anyways!
